zeppelin-zengine/src/test/java/org/apache/zeppelin/helium/HeliumRegistrySerializerTest.java

```diff
@@ -0,0 +1,86 @@
+ * limitations under the License.
+ */
+package org.apache.zeppelin.helium;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class HeliumRegistrySerializerTest {
+
+    private String HELIUM_PACKAGE_JSON_1 = "{\n" +
+            "  \"name\" : \"[organization.A].[name.A]\",\n" +
+            "  \"description\" : \"Description-A\",\n" +
+            "  \"artifact\" : \"groupId:artifactId:version\",\n" +
+            "  \"className\" : \"your.package.name.YourApplicationClass-A\",\n" +
+            "  \"resources\" : [\n" +
+            "    [\"resource.name\", \":resource.class.name\"],\n" +
+            "    [\"alternative.resource.name\", \":alternative.class.name\"]\n" +
+            "  ],\n" +
+            "  \"icon\" : \"<i class='icon'></i>\"\n" +
+            "}";
+
+    @Test
+    void testDeserialization() throws IOException {
+        // Given
+
+        // When
+        // Define the output file
+        Path dir = Files.createDirectory(Paths.get("./" + UUID.randomUUID().toString()));
+        File newFile = Files.createTempFile(dir, UUID.randomUUID().toString(), ".json").toFile();
+        FileUtils.writeStringToFile(newFile, HELIUM_PACKAGE_JSON_1, StandardCharsets.UTF_8);
+        FileUtils.forceDeleteOnExit(newFile);
+
+        // Write JSON string to file using Apache Commons FileUtils
+        HeliumRegistry registry = new HeliumLocalRegistry("my-registry", dir.toString());
+
+        // Then
+        assertEquals("my-registry", registry.name());
+        assertNotNull(registry.getAll());
+        assertEquals(1, registry.getAll().size());
+
+        HeliumPackage heliumPackage = registry.getAll().stream().findFirst().orElseThrow(RuntimeException::new);
+        assertEquals("[organization.A].[name.A]", heliumPackage.getName());
+        assertEquals("Description-A", heliumPackage.getDescription());
+        assertEquals("groupId:artifactId:version", heliumPackage.getArtifact());
+        assertEquals("your.package.name.YourApplicationClass-A", heliumPackage.getClassName());
+        assertEquals(2, heliumPackage.getResources().length);
+        assertEquals("resource.name", heliumPackage.getResources()[0][0]);
+        assertEquals(":resource.class.name", heliumPackage.getResources()[0][1]);
+        assertEquals("alternative.resource.name", heliumPackage.getResources()[1][0]);
+        assertEquals(":alternative.class.name", heliumPackage.getResources()[1][1]);
+        assertEquals("<i class='icon'></i>", heliumPackage.getIcon());
+
+    }
+
+    private File createHeliumPackage(Path testDir, String JSON) throws IOException {
```

Comment:
This method isn't used. If it's not necessary, how about removing it?

Reply:
Done!